lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp

```diff
@@ -2367,11 +2369,38 @@ size_t DWARFASTParserClang::ParseChildEnumerators(
     }
 
     if (name && name[0] && got_value) {
-      m_ast.AddEnumerationValueToEnumerationType(
+      auto ECD = m_ast.AddEnumerationValueToEnumerationType(
           clang_type, decl, name, enum_value, enumerator_byte_size * 8);
       ++enumerators_added;
+
+      llvm::APSInt InitVal = ECD->getInitVal();
+      // Keep track of the size of positive and negative values.
+      if (InitVal.isUnsigned() || InitVal.isNonNegative()) {
+        // If the enumerator is zero that should still be counted as a positive
+        // bit since we need a bit to store the value zero.
+        unsigned ActiveBits = InitVal.getActiveBits();
+        NumPositiveBits = std::max({NumPositiveBits, ActiveBits, 1u});
```

Comment:
```suggestion
        NumPositiveBits = std::max(InitVal.getActiveBits(), 1u);
```

`NumPositiveBits` is 0 at this point anyway right?

Reply:
The condition checks that the value is just non negative unsigned, so I think the comment means that it could be zero, in which case the max function would prefer `1u` over it.

Reply:
I don't think I can move the entire loop into a separate function, the way `DWARFASTParserClang` and `SemaDecl` iterate over enum values is completely different, so for now I just moved these few lines that determine the updated `NumPositiveBits` `NumNegativeBits`.

Reply:
I left a comment on my latest review on how you might be able to do that. Once we did that I think we're good to land this! Thanks for the patience